src/hydra-eval-jobs/hydra-eval-jobs.cc

```diff
@@ -177,15 +177,14 @@ static void worker(
 
                 DrvInfo::Outputs outputs = drv->queryOutputs();
 
-                if (drv->querySystem() == "unknown")
-                    throw EvalError("derivation must have a 'system' attribute");
+                auto drvContent = state.store->readDerivation(drv->requireDrvPath());
```

Comment:
I'm concerned about how much IO this is going to introduce for a project like Nixpkgs.

Reply:
I would expect them to still be in the page cache, but we could cache parsed derivations on the Nix side to save some context switches and a little bit of computation.

Reply:
Also do we even need to return the system here? I have some doubts that we in fact do.